The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10892701247/job/30226164425) [(plain)](https://github.com/rust-lang/rust/commit/7e2cad3c6ff901b728e07f544172d8ac93c046b3/checks/30226164425/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:fb97a1d6377f6cf2227825318ca4bbde3889e0c420746f5a03ba46a07e9a862725c26a09b9fc49a0d129ebd75935d3f6cd19acf41cc4267a6846fd4aa574b12c:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
failures:

---- [codegen] tests/codegen/repr/transparent-struct-ptr.rs#wasm32 stdout ----

error in revision `wasm32`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-17/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/repr/transparent-struct-ptr.wasm32/transparent-struct-ptr.ll" "/checkout/tests/codegen/repr/transparent-struct-ptr.rs" "--check-prefix=CHECK" "--check-prefix" "wasm32" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/codegen/repr/transparent-struct-ptr.rs:52:11: error: CHECK: expected string not found in input
/checkout/tests/codegen/repr/transparent-struct-ptr.rs:52:11: error: CHECK: expected string not found in input
// CHECK: define{{.*}}void @test_BigS(ptr [[BIGS_RET_ATTRS1:.*]] sret([64 x i8]) [[BIGS_RET_ATTRS2:.*]], ptr [[BIGS_ARG_ATTRS1:.*]] byval([64 x i8]) [[BIGS_ARG_ATTRS2:.*]])
          ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/repr/transparent-struct-ptr.wasm32/transparent-struct-ptr.ll:1:1: note: scanning from here
; ModuleID = 'transparent_struct_ptr.2a7c545615247638-cgu.0'

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/repr/transparent-struct-ptr.wasm32/transparent-struct-ptr.ll
Check file: /checkout/tests/codegen/repr/transparent-struct-ptr.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
          1: ; ModuleID = 'transparent_struct_ptr.2a7c545615247638-cgu.0' 
check:52     X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
          2: source_filename = "transparent_struct_ptr.2a7c545615247638-cgu.0" 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-n32:64-S128-ni:1:10:20" 
          4: target triple = "wasm32-unknown-wasi" 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          5:  
check:52     ~
check:52     ~
          6: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          7: define dso_local void @test_BigS(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
          8: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
          9:  br label %bb1 
         10:  
check:52     ~
check:52     ~
         11: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         12:  br label %bb1 
         13: } 
check:52     ~~
         14:  
check:52     ~
check:52     ~
         15: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         16: define dso_local void @test_TsBigS(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         17: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         18:  br label %bb1 
         19:  
check:52     ~
check:52     ~
         20: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         21:  br label %bb1 
         22: } 
check:52     ~~
         23:  
check:52     ~
check:52     ~
         24: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         25: define dso_local void @test_TuBigS(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         26: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         27:  br label %bb1 
         28:  
check:52     ~
check:52     ~
         29: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         30:  br label %bb1 
         31: } 
check:52     ~~
         32:  
check:52     ~
check:52     ~
         33: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         34: define dso_local void @test_TeBigS(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         35: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         36:  br label %bb1 
         37:  
check:52     ~
check:52     ~
         38: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         39:  br label %bb1 
         40: } 
check:52     ~~
         41:  
check:52     ~
check:52     ~
         42: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         43: define dso_local void @test_BigU(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         44: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         45:  br label %bb1 
         46:  
check:52     ~
check:52     ~
         47: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         48:  br label %bb1 
         49: } 
check:52     ~~
         50:  
check:52     ~
check:52     ~
         51: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         52: define dso_local void @test_TsBigU(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         53: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         54:  br label %bb1 
         55:  
check:52     ~
check:52     ~
         56: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         57:  br label %bb1 
         58: } 
check:52     ~~
         59:  
check:52     ~
check:52     ~
         60: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         61: define dso_local void @test_TuBigU(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         62: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         63:  br label %bb1 
         64:  
check:52     ~
check:52     ~
         65: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         66:  br label %bb1 
         67: } 
check:52     ~~
         68:  
check:52     ~
check:52     ~
         69: ; Function Attrs: nounwind 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
         70: define dso_local void @test_TeBigU(ptr noalias nocapture noundef sret([64 x i8]) align 4 dereferenceable(64) %_0, ptr noalias nocapture noundef align 4 dereferenceable(64) %_1) unnamed_addr #0 { 
         71: start: 
check:52     ~~~~~~~
check:52     ~~~~~~~
         72:  br label %bb1 
         73:  
check:52     ~
check:52     ~
         74: bb1: ; preds = %bb1, %start 
check:52     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         75:  br label %bb1 
         76: } 
check:52     ~~
         77:  
check:52     ~
check:52     ~
         78: attributes #0 = { nounwind "target-cpu"="generic" } 
         79:  
check:52     ~
check:52     ~
         80: !llvm.ident = !{!0} 
         81:  
check:52     ~
check:52     ~
         82: !0 = !{!"rustc version 1.83.0-nightly (fbfd833bc 2024-09-16)"} 
>>>>>>
------------------------------------------


```

</details>

--- a/compiler/rustc_target/src/abi/call/m68k.rs
+++ b/compiler/rustc_target/src/abi/call/m68k.rs
@@ -14,7 +14,7 @@ fn classify_arg<Ty>(arg: &mut ArgAbi<'_, Ty>) {
         return;
     }
     if arg.layout.is_aggregate() {
-        arg.make_indirect_byval(None);
+        arg.pass_by_stack_offset(None);
     } else {
         arg.extend_integer_width_to(32);
     }

--- a/compiler/rustc_target/src/abi/call/mod.rs
+++ b/compiler/rustc_target/src/abi/call/mod.rs
@@ -64,7 +64,7 @@ pub enum PassMode {
     /// (which ensures that padding is preserved and that we do not rely on LLVM's struct layout),
     /// and will use the alignment specified in `attrs.pointee_align` (if `Some`) or the type's
     /// alignment (if `None`). This means that the alignment will not always
-    /// match the Rust type's alignment; see documentation of `make_indirect_byval` for more info.
+    /// match the Rust type's alignment; see documentation of `pass_by_stack_offset` for more info.
     ///
     /// `on_stack` cannot be true for unsized arguments, i.e., when `meta_attrs` is `Some`.
     Indirect { attrs: ArgAttributes, meta_attrs: Option<ArgAttributes>, on_stack: bool },
@@ -681,7 +681,7 @@ impl<'a, Ty> ArgAbi<'a, Ty> {
     /// either in the caller (if the type's alignment is lower than the byval alignment)
     /// or in the callee (if the type's alignment is higher than the byval alignment),
     /// to ensure that Rust code never sees an underaligned pointer.
-    pub fn make_indirect_byval(&mut self, byval_align: Option<Align>) {
+    pub fn pass_by_stack_offset(&mut self, byval_align: Option<Align>) {
         assert!(!self.layout.is_unsized(), "used byval ABI for unsized layout");
         self.make_indirect();
         match self.mode {
@@ -879,8 +879,7 @@ impl<'a, Ty> FnAbi<'a, Ty> {
     {
         if abi == spec::abi::Abi::X86Interrupt {
             if let Some(arg) = self.args.first_mut() {
-                // FIXME(pcwalton): This probably should use the x86 `byval` ABI...
-                arg.make_indirect_byval(None);
+                arg.pass_by_stack_offset(None);
             }
             return Ok(());
         }

--- a/compiler/rustc_target/src/abi/call/wasm.rs
+++ b/compiler/rustc_target/src/abi/call/wasm.rs
@@ -40,7 +40,7 @@ where
     }
     arg.extend_integer_width_to(32);
     if arg.layout.is_aggregate() && !unwrap_trivial_aggregate(cx, arg) {
-        arg.make_indirect_byval(None);
+        arg.make_indirect();
     }
 }
 

--- a/compiler/rustc_target/src/abi/call/x86.rs
+++ b/compiler/rustc_target/src/abi/call/x86.rs
@@ -122,7 +122,7 @@ where
                 align_4
             };
 
-            arg.make_indirect_byval(Some(byval_align));
+            arg.pass_by_stack_offset(Some(byval_align));
         } else {
             arg.extend_integer_width_to(32);
         }

--- a/compiler/rustc_target/src/abi/call/x86_64.rs
+++ b/compiler/rustc_target/src/abi/call/x86_64.rs
@@ -219,7 +219,7 @@ where
                 if is_arg {
                     // The x86_64 ABI doesn't have any special requirements for `byval` alignment,
                     // the type's alignment is always used.
-                    arg.make_indirect_byval(None);
+                    arg.pass_by_stack_offset(None);
                 } else {
                     // `sret` parameter thus one less integer register available
                     arg.make_indirect();

--- a/compiler/rustc_target/src/abi/call/xtensa.rs
+++ b/compiler/rustc_target/src/abi/call/xtensa.rs
@@ -68,7 +68,7 @@ where
     *arg_gprs_left -= needed_arg_gprs;
 
     if must_use_stack {
-        arg.make_indirect_byval(None);
+        arg.pass_by_stack_offset(None);
     } else if is_xtensa_aggregate(arg) {
         // Aggregates which are <= max_size will be passed in
         // registers if possible, so coerce to integers.

--- a/tests/codegen/align-byval.rs
+++ b/tests/codegen/align-byval.rs
@@ -1,10 +1,8 @@
 // ignore-tidy-linelength
-//@ revisions:m68k wasm x86_64-linux x86_64-windows i686-linux i686-windows
+//@ revisions:m68k x86_64-linux x86_64-windows i686-linux i686-windows
 
 //@[m68k] compile-flags: --target m68k-unknown-linux-gnu
 //@[m68k] needs-llvm-components: m68k
-//@[wasm] compile-flags: --target wasm32-unknown-emscripten
-//@[wasm] needs-llvm-components: webassembly
 //@[x86_64-linux] compile-flags: --target x86_64-unknown-linux-gnu
 //@[x86_64-linux] needs-llvm-components: x86
 //@[x86_64-windows] compile-flags: --target x86_64-pc-windows-msvc
@@ -15,7 +13,7 @@
 //@[i686-windows] needs-llvm-components: x86
 
 // Tests that `byval` alignment is properly specified (#80127).
-// The only targets that use `byval` are m68k, wasm, x86-64, and x86.
+// The only targets that use `byval` are m68k, x86-64, and x86.
 // Note also that Windows mandates a by-ref ABI here, so it does not use byval.
 
 #![feature(no_core, lang_items)]
@@ -112,9 +110,6 @@ pub unsafe fn call_na1(x: NaturalAlign1) {
     // m68k: [[ALLOCA:%[a-z0-9+]]] = alloca [2 x i8], align 1
     // m68k: call void @natural_align_1({{.*}}byval([2 x i8]) align 1{{.*}} [[ALLOCA]])
 
-    // wasm: [[ALLOCA:%[a-z0-9+]]] = alloca [2 x i8], align 1
-    // wasm: call void @natural_align_1({{.*}}byval([2 x i8]) align 1{{.*}} [[ALLOCA]])
-
     // x86_64-linux: call void @natural_align_1(i16
 
     // x86_64-windows: call void @natural_align_1(i16
@@ -133,7 +128,6 @@ pub unsafe fn call_na2(x: NaturalAlign2) {
     // CHECK: start:
 
     // m68k-NEXT: call void @natural_align_2
-    // wasm-NEXT: call void @natural_align_2
     // x86_64-linux-NEXT: call void @natural_align_2
     // x86_64-windows-NEXT: call void @natural_align_2
 
@@ -204,8 +198,6 @@ pub unsafe fn call_fa16(x: ForceAlign16) {
 extern "C" {
     // m68k: declare void @natural_align_1({{.*}}byval([2 x i8]) align 1{{.*}})
 
-    // wasm: declare void @natural_align_1({{.*}}byval([2 x i8]) align 1{{.*}})
-
     // x86_64-linux: declare void @natural_align_1(i16)
 
     // x86_64-windows: declare void @natural_align_1(i16)
@@ -216,8 +208,6 @@ extern "C" {
     fn natural_align_1(x: NaturalAlign1);
 
     // m68k: declare void @natural_align_2({{.*}}byval([34 x i8]) align 2{{.*}})
-
-    // wasm: declare void @natural_align_2({{.*}}byval([34 x i8]) align 2{{.*}})
 
     // x86_64-linux: declare void @natural_align_2({{.*}}byval([34 x i8]) align 2{{.*}})
 
@@ -232,8 +222,6 @@ extern "C" {
 
     // m68k: declare void @force_align_4({{.*}}byval([20 x i8]) align 4{{.*}})
 
-    // wasm: declare void @force_align_4({{.*}}byval([20 x i8]) align 4{{.*}})
-
     // x86_64-linux: declare void @force_align_4({{.*}}byval([20 x i8]) align 4{{.*}})
 
     // x86_64-windows: declare void @force_align_4(
@@ -247,8 +235,6 @@ extern "C" {
 
     // m68k: declare void @natural_align_8({{.*}}byval([24 x i8]) align 4{{.*}})
 
-    // wasm: declare void @natural_align_8({{.*}}byval([24 x i8]) align 8{{.*}})
-
     // x86_64-linux: declare void @natural_align_8({{.*}}byval([24 x i8]) align 8{{.*}})
 
     // x86_64-windows: declare void @natural_align_8(
@@ -261,8 +247,6 @@ extern "C" {
     fn natural_align_8(x: NaturalAlign8);
 
     // m68k: declare void @force_align_8({{.*}}byval([24 x i8]) align 8{{.*}})
-
-    // wasm: declare void @force_align_8({{.*}}byval([24 x i8]) align 8{{.*}})
 
     // x86_64-linux: declare void @force_align_8({{.*}}byval([24 x i8]) align 8{{.*}})
 
@@ -279,8 +263,6 @@ extern "C" {
 
     // m68k: declare void @lower_fa8({{.*}}byval([24 x i8]) align 4{{.*}})
 
-    // wasm: declare void @lower_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
-
     // x86_64-linux: declare void @lower_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
 
     // x86_64-windows: declare void @lower_fa8(
@@ -293,8 +275,6 @@ extern "C" {
     fn lower_fa8(x: LowerFA8);
 
     // m68k: declare void @wrapped_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
-
-    // wasm: declare void @wrapped_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
 
     // x86_64-linux: declare void @wrapped_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
 
@@ -311,8 +291,6 @@ extern "C" {
 
     // m68k: declare void @transparent_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
 
-    // wasm: declare void @transparent_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
-
     // x86_64-linux: declare void @transparent_fa8({{.*}}byval([24 x i8]) align 8{{.*}})
 
     // x86_64-windows: declare void @transparent_fa8(
@@ -327,8 +305,6 @@ extern "C" {
     fn transparent_fa8(x: TransparentFA8);
 
     // m68k: declare void @force_align_16({{.*}}byval([80 x i8]) align 16{{.*}})
-
-    // wasm: declare void @force_align_16({{.*}}byval([80 x i8]) align 16{{.*}})
 
     // x86_64-linux: declare void @force_align_16({{.*}}byval([80 x i8]) align 16{{.*}})
 

